## https://github.com/suhrig/arriba
p1 <- InputParam(id = "align", type = "File", prefix = "-x")
p2 <- InputParam(id = "out", type = "string", prefix = "-o")
p3 <- InputParam(id = "dout", type = "string", prefix = "-O")
p4 <- InputParam(id = "genome", type = "File", prefix = "-a", secondaryFiles = ".fai")
p5 <- InputParam(id = "gtf", type = "File", prefix = "-g")
p6 <- InputParam(id = "blacklist", type = "File", prefix = "-b")
p7 <- InputParam(id = "known", type = "File", prefix = "-k")
p8 <- InputParam(id = "tag", type = "File", prefix = "-t")
p9 <- InputParam(id = "protein", type = "File", prefix = "-p")
o1 <- OutputParam(id = "fout", type = "File", glob = "$(inputs.out)")
o2 <- OutputParam(id = "fOut", type = "File", glob = "$(inputs.dout)" )
req1 <- requireDocker(searchContainer("arriba")$container[1])
arriba <- cwlProcess(baseCommand = "arriba",
                     requirements = list(req1),
                     inputs = InputParamList(p1, p2, p3, p4, p5, p6, p7, p8, p9),
                     outputs = OutputParamList(o1, o2))


arriba <- addMeta(
    arriba,
    label = "arriba",
    doc = "Arriba is a fast tool to search for aberrant transcripts such as gene fusions. It is based on chimeric alignments found by the STAR RNA-Seq aligner.",
    inputLabels = c("align","out","dout","genome","gtf","blacklist","known","tag","protein"),
    inputDocs = c("File in SAM/BAM/CRAM format with main alignments as generated by STAR (Aligned.out.sam). Arriba extracts candidate reads from this file.","Name of output file with fusions that have passed all filters.","Name of output file with fusions that were discarded due to filtering.","FastA file with genome sequence (assembly). The file may be gzip-compressed. An index with the file extension .fai must exist only if CRAM files are processed.","GTF file with gene annotation. The file may be gzip-compressed.","File containing blacklisted events (recurrent artifacts and transcripts observed in healthy tissue).","File containing known/recurrent fusions. Some cancer entities are often characterized by fusions between the same pair of genes. In order to boost sensitivity, a list of known fusions can be supplied using this parameter. The list must contain two columns with the names of the fused genes, separated by tabs.","File containing list of user-defined tags","Protein domain annotation File"),
    outputLabels = c("fout","fOut"),
    outputDocs = c("Output file with fusions that have passed all filters.","Output file with fusions that were discarded due to filtering."),
    extensions = list(
        author = "rworkflow team",
        date = "09-08-24",
        url = "https://github.com/suhrig/arriba",
        example = paste()
    )
)
